### PR TITLE
Move MeasurementResult into _typing to fix circular dependency in #1449

### PR DESCRIPTION
**Pull request by amirebrahimi**
## Description

This is a preliminary PR to #1449 to fix a circular dependency with imports.

---

### License

- [X] I license this contribution under the terms of the GNU GPL, version 3 and grant Unitary Fund the right to provide additional permissions as described in section 7 of the GNU GPL, version 3.

Before opening the PR, please ensure you have completed the following where appropriate.
- [ ] I added unit tests for new code.
- [ ] I used [type hints](https://www.python.org/dev/peps/pep-0484/) in function signatures.
- [ ] I used [Google-style](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods) docstrings for functions.
- [ ] I [updated the documentation](../blob/master/docs/CONTRIBUTING_DOCS.md) where relevant.
- [ ] Added myself / the copyright holder to the AUTHORS file

### CODE REVIEW

**Comment by amirebrahimi** [comment]
Looks like all tests are passing. The code coverage is from something I didn't change, but I think triggered because I removed a line of import code.

**Review by andreamari — APPROVED** [review]
Thanks @amirebrahimi! LGTM

File: .all-contributorsrc [hunk 1]
```diff
@@ -355,6 +355,17 @@
...
+    {
+      "login": "amirebrahimi",
+      "name": "Amir Ebrahimi",
+      "avatar_url": "https://avatars.githubusercontent.com/u/904110?v=4",
+      "profile": "https://github.com/amirebrahimi/",
+      "contributions": [
+        "doc",
+        "test",
+        "code"
+      ]
```
[the remark this answers is not in the document]
Comment (amirebrahimi) [reply .all-contributorsrc]: I committed the suggested changes, so I'm not sure if you'll have to make the request to the bot again. Perhaps that will happen delayed from the last commit.

[the remark this answers is not in the document]
Comment (andreamari) [reply .all-contributorsrc]: @allcontributors please add @amirebrahimi for code, test, doc.

(I think there was a typo in all-contributors)

[the remark this answers is not in the document]
Comment (natestemen) [reply .all-contributorsrc]: I wonder if the bot doesn't work in pull request review comments?

[the remark this answers is not in the document]
Comment (andreamari) [reply .all-contributorsrc]: > I wonder if the bot doesn't work in pull request review comments?

You are right! I added a new comment in this [issue](https://github.com/unitaryfund/mitiq/issues/815#issuecomment-1244059844) and it worked.

**Pull request merged**